src/config/gobjects.rs

```diff
@@ -143,8 +146,25 @@ fn parse_object(toml_object: &Value, concurrency: library::Concurrency) -> GObje
         .and_then(|v| v.as_str())
         .and_then(|v| v.parse().ok())
         .unwrap_or(concurrency);
+    let ref_mode = toml_object
+        .lookup("ref_mode")
+        .and_then(|v| v.as_str())
+        .and_then(|v| match v {
+            "none" => Some(ref_mode::RefMode::None),
+            "ref" => Some(ref_mode::RefMode::ByRef),
+            "ref-mut" => Some(ref_mode::RefMode::ByRefMut),
+            "ref-immut" => Some(ref_mode::RefMode::ByRefImmut),
+            "ref-fake" => Some(ref_mode::RefMode::ByRefFake),
+            _ => None,
+        });
```

Comment:
Please move this match out this function. Just private function in this module IMHO will do as `analysis/ref_mode.rs` use this module.